### DISCUSSION TITLE
Support CTest without CMake

### DISCUSSION THREAD

**Discussion by kalvdans** [microsoft/vscode-cmake-tools#4131] [Ideas]
We have a Python project with no compiled code, and no need for cmake. But we still like to use ctest as the test driver. It works find from the command-line, and have the list of tests in CTestTestfile.cmake (as [documented](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Testing%20With%20CMake%20and%20CTest.html#testing-using-ctest)).

In order to get them into vscode test explorer, two hacks are needed to trick [cmake-test-explorer](https://github.com/fredericbonnet/cmake-test-explorer) to believe the source directory is a build directory:
* Create a CMakeCache.txt with a single line: `CMAKE_CTEST_COMMAND:INTERNAL=ctest`
* Change configuration: "cmakeExplorer.buildDir": "."

Would it be possible to make vscode-cmake-tools discover a testable directory by the existance of CTestTestfile.cmake?

**Comment by gcampbell-msft** [comment]
@kalvdans This sounds like a feature request, could you create an issue on the Issues pane so that we can properly track this? Thanks!

**Reply by kalvdans** [reply]
Sure! Opened an issue at https://github.com/microsoft/vscode-cmake-tools/issues/4169
